### ISSUE TITLE
Enum handling in high level API

### ISSUE THREAD

**Issue by kopp** [typed-argparse/typed-argparse#36]
It looks like no special handling of enums is necessary in the high level API go get them parsed correctly (as opposed to [`get_choices_from`](https://typed-argparse.github.io/typed-argparse/low_level_api/#convenience-functionality-to-map-literalenum-to-choices)).

I currently overwrite the `__str__(self)` method for my enums, to get the values _displayed_ correctly in the output of `--help`, see the example below.

Is this the intended way to handle enums in the high level API?

```
from enum import Enum
from typing import List, Optional

import typed_argparse as tap


class Adjective(Enum):
    cool = "quite_cool"
    awesome = "awesome"
    over_the_top = 3

    def __str__(self):
        return str(self.value)


# 1. Argument definition
class Args(tap.TypedArgs):
    my_arg: str = tap.arg(help="some help")
    number_a: int = tap.arg(default=42, help="some help")
    number_b: Optional[int] = tap.arg(help="some help")
    verbose: bool = tap.arg(help="some help")
    names: List[str] = tap.arg(help="some help")
    how_cool_is_this: Adjective = tap.arg(help="how cool typed_argparse is")


# 2. Business logic
def runner(args: Args):
    print(f"Running my app with args:\n{args}")


# 3. Bind argument definition + business logic & run
def main() -> None:
    tap.Parser(Args).bind(runner).run()


if __name__ == "__main__":
    main()

```

The output of `--help` is

```
...
  --how-cool-is-this {quite_cool,awesome,3}
                        how cool typed_argparse is
```

and without overwriting `__str__` it is

```
...
  --how-cool-is-this {Adjective.cool,Adjective.awesome,Adjective.over_the_top}
                        how cool typed_argparse is
```

**Comment by bluenote10** [comment]
In general yes, but you'll probably have to implement both `__str__` and `__repr__` because argparse uses both internally (one for the help message and the other one for the error message if the value was wrong 😕 ). Failing to do so isn't an error or so, just not-so-nice messages for the user.

To avoid having to implement `__str__` and `__repr__` at all, I'd actually recommend to use "mixing enums" like `class MyStringLikeEnum(str, Enum)` or `class MyIntLikeEnum(int, Enum)`. This has the further benefit the one doesn't need to use `.value` everywhere. The `MyStringLikeEnum.FOOBAR` simply is a string already, which simplifies the enum usage in general.

**Comment by kopp** [comment]
If I modify the above example to use

```
class Adjective(str, Enum):
    cool = "quite_cool"
    awesome = "awesome"
```

then `--help` will report

```
...
  --how-cool-is-this {Adjective.cool,Adjective.awesome}
                        how cool typed_argparse is
```

I will still have to implement `__str__`, but can simply `return self` instead of `self.value`.

**Comment by bluenote10** [comment]
Should be closed by #37
